ducktape/cluster/node_container.py

```diff
@@ -23,8 +27,23 @@ class InsufficientResourcesError(Exception):
     pass
 
 
+class InsufficientHealthyNodesError(InsufficientResourcesError):
+
+    def __init__(self, bad_nodes, *args):
+        self.bad_nodes = bad_nodes
+        super().__init__(*args)
+
+
+@dataclass
+class RemoveSpecResult:
+    good_nodes: List = field(default_factory=list)
+    bad_nodes: List = field(default_factory=list)
+
```

Comment:
cool!

Reply:
but then why would you even need a result spec? can simply do `return good_nodes, bad_nodes`

Reply:
did exactly that, seems more pythonic and occam-ic :)

Reply:
well sometimes you want both i guess. if you only want the bad nodes you can do `get_results().bad_nodes` but if you want both you can do `good_nodes, bad_nodes = get_results()`  but hey whatever you'd like here to be honest not super important

Reply:
yeah, I see your point. It's expressiveness vs simplicity I think. In this case I think an unnamed result is a bit more 'pythonic', but it can be argued about for sure. You're right about it being a small detail, so I'll leave it unnamed :)